### PR TITLE
further Clicks are implemented

### PR DESCRIPTION
**Pull request by noubar**
further Clicks are implemented
click_element_identifier argument of click close changed to be an optional argument

### CODE REVIEW

**Comment by Nepitwin** [comment]
@noubar 

Forget to update changelog

**Comment by noubar** [comment]
i did not forget, the changelog is updated within the previous commit. The implementation took me two PRs because i needed your help in between.

File: atests/Mouse.robot [hunk 1]
```diff
@@ -27,6 +30,13 @@ ${PopupToggle2_BUTTON}     ${MAIN_WINDOW_SIMPLE_CONTROLS}/Button[@AutomationId='
 ${ENABLE_BUTTON}           ${MAIN_WINDOW_SIMPLE_CONTROLS}/Button[@AutomationId='EnableButton']
 ${READYTOTAKEOFF_TEXT}     ${MAIN_WINDOW_SIMPLE_CONTROLS}/Button[@AutomationId='EnableButton']/Text[@Name='Ready to take off']
 ${TOGGLE_BUTTON}           ${MAIN_WINDOW_SIMPLE_CONTROLS}/Button[@AutomationId='ToggleButton']
+${XPATH_COMBO_BOX}         ${MAIN_WINDOW_SIMPLE_CONTROLS}/ComboBox[@AutomationId='NonEditableCombo']
+${CONTEXTMENU_BUTTON}      ${MAIN_WINDOW_SIMPLE_CONTROLS}/Button[@AutomationId='ContextMenu']
+${XPATH_COMBO_BOX_ITEM}    ${XPATH_COMBO_BOX}/ListItem[@Name='Item 3']/Text[@Name='Item 3']
+${XPATH_TREE_PARENT}       ${MAIN_WINDOW_COMPLEX_CONTROLS}/Pane/Group[@Name='Tree']/Tree[@AutomationId='treeView1']/TreeItem[@Name='Lvl1 a']/Text[@Name='Lvl1 a']
+${XPATH_TREE_CHILD}        ${MAIN_WINDOW_COMPLEX_CONTROLS}/Pane/Group[@Name='Tree']/Tree[@AutomationId='treeView1']/TreeItem[@Name='Lvl1 a']/TreeItem[@Name='Lvl2 a']/Text[@Name='Lvl2 a']
+
+/Window[@Name='FlaUI WPF Test App']/Window[@ClassName='Popup']/Menu[@ClassName='ContextMenu']/MenuItem[@Name='Context 2']
```
Comment (Nepitwin) [inline atests/Mouse.robot:39]: Remove it?

Missing variable usage?

File: src/FlaUILibrary/keywords/mouse.py [hunk 2]
```diff
@@ -249,7 +249,77 @@ def click_open(self, click_element_identifier: str, open_element_identifier: str
...
+        | open_element_identifier | string | XPath identifier from element to be opened after double clicking click_element |
+        | focus_element_identifier_before_click | string | XPath identifier from element to be focused before double clicking click_element |
+        | focus_element_identifier_after_open | string | XPath identifier from element to be focused after openning open_element |
+        | max_repeat | int | Maximum number of tries |
+        | timeout_between_repeates | int | wait time in milli seconds in between every try |
+        | ignore_if_already_open | bool | the keyword will not be executed if excepted element is already open |
+        | msg | string | Custom error message |
+
+        Examples:
+        | Click Open  <XPATH>  <XPATH>  |
```
Comment (Nepitwin) [inline src/FlaUILibrary/keywords/mouse.py:276]: Double Click Open

File: src/FlaUILibrary/keywords/mouse.py [hunk 3]
```diff
@@ -249,7 +249,77 @@ def click_open(self, click_element_identifier: str, open_element_identifier: str
...
+        | open_element_identifier | string | XPath identifier from element to be opened after right clicking click_element |
+        | focus_element_identifier_before_click | string | XPath identifier from element to be focused before right clicking click_element |
+        | focus_element_identifier_after_open | string | XPath identifier from element to be focused after openning open_element |
+        | max_repeat | int | Maximum number of tries |
+        | timeout_between_repeates | int | wait time in milli seconds in between every try |
+        | ignore_if_already_open | bool | the keyword will not be executed if excepted element is already open |
+        | msg | string | Custom error message |
+
+        Examples:
+        | Click Open  <XPATH>  <XPATH>  |
```
Comment (Nepitwin) [inline src/FlaUILibrary/keywords/mouse.py:311]: Right Click Open

File: src/FlaUILibrary/keywords/mouse.py [hunk 4]
```diff
@@ -276,9 +346,85 @@ def click_close(self, click_element_identifier: str, close_element_identifier: s
         | Click Open  <XPATH>  <XPATH>  |
```
Comment (Nepitwin) [inline src/FlaUILibrary/keywords/mouse.py:346]: Click Close

File: src/FlaUILibrary/keywords/mouse.py [hunk 5]
```diff
@@ -276,9 +346,85 @@ def click_close(self, click_element_identifier: str, close_element_identifier: s
...
+        | close_element_identifier | string | XPath identifier from element to be closed after double clicking click_element if not given the click_element_identifier will be set as default |
+        | focus_element_identifier_before_click | string | XPath identifier from element to be focused before double clicking click_element |
+        | focus_element_identifier_after_close | string | XPath identifier from element to be focused after openning close_element |
+        | max_repeat | int | Maximum number of tries |
+        | timeout_between_repeates | int | wait time in milli seconds in between every try |
+        | ignore_if_already_open | bool | the keyword will not be executed if excepted element is already closed |
+        | msg        | string | Custom error message          |
+
+        Examples:
+        | Click Open  <XPATH>  <XPATH>  |
```
Comment (Nepitwin) [inline src/FlaUILibrary/keywords/mouse.py:383]: Double Click Close

File: src/FlaUILibrary/keywords/mouse.py [hunk 6]
```diff
@@ -276,9 +346,85 @@ def click_close(self, click_element_identifier: str, close_element_identifier: s
...
+        | close_element_identifier | string | XPath identifier from element to be closed after right clicking click_element if not given the click_element_identifier will be set as default |
+        | focus_element_identifier_before_click | string | XPath identifier from element to be focused before right clicking click_element |
+        | focus_element_identifier_after_close | string | XPath identifier from element to be focused after openning close_element |
+        | max_repeat | int | Maximum number of tries |
+        | timeout_between_repeates | int | wait time in milli seconds in between every try |
+        | ignore_if_already_open | bool | the keyword will not be executed if excepted element is already closed |
+        | msg        | string | Custom error message          |
+
+        Examples:
+        | Click Open  <XPATH>  <XPATH>  |
```
Comment (Nepitwin) [inline src/FlaUILibrary/keywords/mouse.py:420]: Right Click Close

**Comment by Nepitwin** [comment]
@noubar 

Pls recheck by Keyword Mouse if Examples Are correct keywords. Common documentation is wrong keyword example

**Review by Nepitwin — CHANGES_REQUESTED** [review]

**Review by Nepitwin — APPROVED** [review]

**Pull request merged**